fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/adapters/EventExpandableListViewAdapter.java

```diff
@@ -38,311 +40,376 @@
-   * @param childPosition the position of the child (for which the View is returned) within the
-   *     group
-   * @param isLastChild Whether the child is the last child within the group
-   * @param convertView the old view to reuse, if possible. You should check that this view is
-   *     non-null and of an appropriate type before using. If it is not possible to convert this
-   *     view to display the correct data, this method can create a new view. It is not guaranteed
-   *     that the convertView will have been previously created by {@link #getChildView(int, int,
-   *     boolean, View, ViewGroup)}.
-   * @param parent the parent that this view will eventually be attached to
-   * @return the View corresponding to the child at the specified position
-   */
-  @Override
-  public View getChildView(
-          int groupPosition,
-          int childPosition,
-          boolean isLastChild,
-          View convertView,
-          ViewGroup parent) {
-
-    // TODO : For the moment, events are displayed the same if user is attendee or organizer,
-    // in the future it could be nice to have a pencil icon to allow organizer to modify an event
-
-    LayoutRollCallEventBinding binding;
-
-    if (convertView == null) {
-      LayoutInflater inflater = LayoutInflater.from(parent.getContext());
-      binding = LayoutRollCallEventBinding.inflate(inflater, parent, false);
-    } else {
-      binding = DataBindingUtil.getBinding(convertView);
+    /**
+     * Gets a View that displays the data for the given child within the given group.
+     *
+     * @param groupPosition the position of the group that contains the child
+     * @param childPosition the position of the child (for which the View is returned) within the
+     *                      group
+     * @param isLastChild   Whether the child is the last child within the group
+     * @param convertView   the old view to reuse, if possible. You should check that this view is
+     *                      non-null and of an appropriate type before using. If it is not possible to convert this
+     *                      view to display the correct data, this method can create a new view. It is not guaranteed
+     *                      that the convertView will have been previously created by {@link #getChildView(int, int,
+     *                      boolean, View, ViewGroup)}.
+     * @param parent        the parent that this view will eventually be attached to
+     * @return the View corresponding to the child at the specified position
+     */
+    @Override
+    public View getChildView(
+            int groupPosition,
+            int childPosition,
+            boolean isLastChild,
+            View convertView,
+            ViewGroup parent) {
+
+        Event event = ((Event) getChild(groupPosition, childPosition));
+        EventCategory category = (EventCategory) getGroup(groupPosition);
+        LayoutEventBinding layoutEventBinding;
```

Comment:
I think it is better to create a different binding for each event and not use LayoutEventBinding at all. What you can do is to put all the code from creating the binding until returning the view in the functions setUpElectionElement/setUpRollCallElement. Then in the if branches you call the setUp method and return what this method returns.

Reply:
Thank You I was able to fix the roll call creation problem with the Bindings .